contracts/CollectionParty.sol

```diff
@@ -66,21 +73,26 @@ contract PartyBuy is Party {
 
     function initialize(
         address _nftContract,
-        uint256 _tokenId,
         uint256 _maxPrice,
         uint256 _secondsToTimeout,
+        address[] calldata _deciders,
         Structs.AddressAndAmount calldata _split,
         Structs.AddressAndAmount calldata _tokenGate,
         string memory _name,
         string memory _symbol
     ) external initializer {
-        // validate maxPrice
-        require(_maxPrice > 0, "PartyBuy::initialize: must set price higher than 0");
         // initialize & validate shared Party variables
-        __Party_init(_nftContract, _tokenId, _split, _tokenGate, _name, _symbol);
+        __Party_init(_nftContract, _split, _tokenGate, _name, _symbol);
         // set PartyBuy-specific state variables
         expiresAt = block.timestamp + _secondsToTimeout;
         maxPrice = _maxPrice;
+        // attempt to calculate maximum contributions to ensure this value won't overflow later
+        getMaximumContributions();
```

Comment:
Does this actually catch it?  Probably does, but since its a view function that doesn't store its result it could be safely optimized out by the compiler

Reply:
hmmm i have a test for it and i think the tests should be running the optimized version as well